File: java/get-multiple-keys/src/main/java/com/ionic/samples/GetMultipleKeys.java
/*
 * (c) 2018 Ionic Security Inc.
 * By using this code, I agree to the Terms & Conditions (https://dev.ionic.com/use.html)
 * and the Privacy Policy (https://www.ionic.com/privacy-notice/).
 * 
 * Developed with Ionic Java SDK 2.1.0
 */

package com.ionic.samples;

import com.ionic.sdk.device.profile.persistor.DeviceProfilePersistorPassword;
import com.ionic.sdk.agent.Agent;
import com.ionic.sdk.agent.request.getkey.GetKeysRequest;
import com.ionic.sdk.agent.request.getkey.GetKeysResponse;
import com.ionic.sdk.error.IonicException;
import java.util.List;
import javax.xml.bind.DatatypeConverter;

public class GetMultipleKeys
{
    public static void main(String[] args)
    {
        // initialize agent
        Agent agent = new Agent();
        try {
            String persistorPath = System.getProperty("user.home") + "/.ionicsecurity/profiles.pw";
            String persistorPassword = System.getenv("IONIC_PERSISTOR_PASSWORD");

            DeviceProfilePersistorPassword persistor = new DeviceProfilePersistorPassword(persistorPath);
            persistor.setPassword(persistorPassword);

            agent.initialize(persistor);
        } catch(IonicException e) {
            System.out.println(e.getMessage());
            System.exit(1);
        }
        
        String keyId1 = "HVzG3oKSL7A";
        String keyId2 = "HVzG4fqePwk";

        // get multiple keys
        GetKeysRequest request = new GetKeysRequest();
        request.add(keyId1);
        request.add(keyId2);
        List<GetKeysResponse.Key> keys = null;
        try {
            keys = agent.getKeys(request).getKeys();
        } catch(IonicException e) {
            System.out.println(e.getMessage());
            System.exit(1);
        }

        // display fetched keys
        for (GetKeysResponse.Key key : keys) {
            System.out.println("---");
            System.out.println("KeyId        : " + key.getId());
            System.out.println("KeyBytes     : " + DatatypeConverter.printHexBinary(key.getKey()));
            System.out.println("FixedAttrs   : " + key.getAttributesMap());
            System.out.println("MutableAttrs : " + key.getMutableAttributes());
        }
    }
}

File: javascript/get-multiple-keys/getMultipleKeys.js
/*
 * (c) 2019-2020 Ionic Security Inc.
 * By using this code, I agree to the Terms & Conditions (https://dev.ionic.com/use.html)
 * and the Privacy Policy (https://www.ionic.com/privacy-notice/).
 */

const appData = {
  appId: 'ionic-js-samples',
  userId: 'developer',
  userAuth: 'password123'
}

const main = async () => {
  const keyId1 = 'HVzG5uKl3yE'
  const keyId2 = 'HVzG3AJoHQU'
  const keyId3 = 'HVzG52Kj3to'

  // initialize agent
  const agent = new window.IonicSdk.ISAgent()
  await agent.loadUser(appData).catch((error) => {
    console.log('Error loading profile: ', error)
  })

  // Set the app metadata.
  let response = await agent.setMetadata({
    'ionic-application-name': 'JavaScript getMultipleKeys',
    'ionic-application-verison': '1.1.0',
  })

  // get multiple keys
  response = await agent.getKeys({
    keyIds: [keyId1, keyId2, keyId3]
  }).catch((error) => {
    console.log('Error Creating Key: ', error)
  })
  const keys = response.keys;

  // display fetched keys
  console.log('');
  keys.forEach((key) => {
    console.log('---');
    console.log('KeyId             : ', key.keyId);
    console.log('KeyBytes          : ', key.key);
    console.log('FixedAttributes   : ', JSON.stringify(key.attributes, null, 0));
    console.log('MutableAttributes : ', JSON.stringify(key.mutableAttributes, null, 0));
  })
}

main()
